numba/tests/test_dyn_array.py

```diff
@@ -0,0 +1,362 @@
+from __future__ import print_function, absolute_import, division
+
+import sys
+import numpy as np
+import threading
+
+from numba import unittest_support as unittest
+from numba import njit
+from numba import utils
+
+
+
+nrtjit = njit(nrt=True)
```

Comment:
I think having `nrt=True` to be the default and allowing it to be disabled could be helpful for numba developers for testing and debugging.  We can make it `_nrt` to suggesting it is internal usage.  What do you think?